api/src/controllers/user.js

```diff
@@ -85,124 +86,142 @@ exports.list = (req, res) => {
 	}
 };
 
-exports.delete = (req, res) => {
+exports.delete = async (req, res) => {
 	// authorize access to the authenticated user model only
 	if (req.params.userId !== req.user.sub) {
```

Comment:
why do we even use a parameter for this? just use req.user.sub :)

Reply:
Is your comment about the 'if' statement on line 91?

Without this check any authenticated user can delete another user's record. I don't think we want that?

The resource to be deleted is identified in the URL, e.g. `/user/dwight`. I also don't think it makes sense that a client would send a DELETE identifying a user resource and then blow away the unrelated user record associated with the authenticated JWT.

Reply:
my point is: what is the reason to have a parameter (`req.params.userId`) if the only allowed value is pre-determined (`req.user.sub`)?